signing_service/signing_service/signing_service.py

```diff
@@ -135,6 +135,37 @@ def _validate_arguments(self):
 
 
 def _parse_arguments() -> argparse.Namespace:
+    def secret_provider_factory(read_command_line=False, env_variable_name=None, use_file=False, base64_convert=False):
```

Comment:
As it turned out when we talked last time - it's not really a factory. More like factory factory :)

Maybe call it `make_secret_provider_factory()`?

Reply:
I remember, but I didn't know which name will be the best :P

Reply:
There's really no good name for it :) But `make_...` at least hints at what's going on inside.